src/components/Table/utils/RowStates.ts

```diff
@@ -30,81 +40,117 @@ export class RowStates {
    */
   setRows(rows: GridDataRow<any>[]): void {
     const existing = new Set(this.map.values());
+    const states = this;
     const map = this.map;
 
-    function addRowAndChildren(parent: RowState | undefined, row: GridDataRow<any>): RowState {
+    function addRowAndChildren(row: GridDataRow<any>): RowState {
       // This should really be kind+id, but a lot of our lookups just use id
       const key = row.id;
       let state = map.get(key);
       if (!state) {
-        state = new RowState(parent, row);
+        state = new RowState(states, row);
         map.set(key, state);
       } else {
-        state.parent = parent;
         state.row = row;
-        state.wasRemoved = false;
+        state.removed = false;
         existing.delete(state);
       }
-      state.children = row.children?.map((child) => addRowAndChildren(state, child));
+      state.children = row.children?.map((child) => addRowAndChildren(child));
       return state;
     }
 
     // Probe for the header row, so we can create it as a root RowState, even
     // though we don't require the user to model their GridDataRows that way.
-    const header = rows.find((r) => r.kind === HEADER);
-    const headerState = header ? addRowAndChildren(undefined, header) : undefined;
+    const headerRow = rows.find((r) => r.kind === HEADER);
+    this.header = headerRow ? addRowAndChildren(headerRow) : undefined;
```

Comment:
All of this code is basically the same, just it's now assigning the `this.header` and `this.topRows` fields.